handler.go

```diff
@@ -332,11 +333,36 @@ func (ca *Cagent) GetAllMeasurements() (MeasurementsMap, error) {
 
 	if len(errs) == 0 {
 		measurements["cagent.success"] = 1
+	} else {
+		measurements["cagent.success"] = 0
+	}
 
-		return measurements, nil
+	// measurements fetched below should not affect cagent.success
+	ca.smartInit.Do(func() {
+		if ca.Config.SMARTMonitoring {
+			if err := smart.DetectTools(); err == nil {
+				ca.smartAvailable = true
+			} else {
+				ca.smartAvailable = false
+				log.Error(err.Error())
+			}
+		} else {
+			ca.smartAvailable = false
+		}
+	})
+
+	if ca.smartAvailable {
+		res, err := smart.Parse()
+		measurements = measurements.AddInnerWithPrefix("smartmon", res)
+
+		for _, e := range err {
+			errs = append(errs, e.Error())
+		}
 	}
 
-	measurements["cagent.success"] = 0
+	if len(errs) == 0 {
```

Comment:
I would check against `len(errs) != 0` and in case return `measurements, errors.New(strings.Join(errs, "; "))`

It's just a nuance, but usually we check for error/special cases and then return. The final return should be the success case.

Reply:
done

Reply:
@troian Why do you mark this as resolved if no change happened?